### PR TITLE
Doc | Sync sample changes into snippet doc

### PR DESCRIPTION
**Pull request by David-Engel**
Sync samples changes from #2821 into snippet XML.

### CODE REVIEW

**Comment by DavoudEshtehari** [comment]
There is no reference to `CustomDeviceCodeFlowAzureAuthenticationProvider.cs` file. It looks like a leftover of the [#2822](https://github.com/dotnet/SqlClient/pull/2822) change. Could you remove this file, plz?

**Review by DavoudEshtehari — APPROVED** [review]

**Comment by David-Engel** [comment]
> There is no reference to `CustomDeviceCodeFlowAzureAuthenticationProvider.cs` file. It looks like a leftover of the [#2822](https://github.com/dotnet/SqlClient/pull/2822) change. Could you remove this file, plz?

No. Samples are still referenced by the docs in the sql-docs repo. Removing the files breaks their build.

**Comment by DavoudEshtehari** [comment]
Oh, so we should partially revert #2822.

**Comment by David-Engel** [comment]
> Oh, so we should partially revert #2822.

Already been done. #2835

File: doc/snippets/Microsoft.Data.SqlClient/SqlAuthenticationProvider.xml [hunk 1]
```diff
@@ -9,10 +9,12 @@
         The following example demonstrates implementing a custom SqlAuthenticationProvider and providing the same to SqlClient for overriding Device Code Flow authentication mode:
         <code language="c#">
           using System;
+          using System.Collections.Generic;
+          using System.Linq;
```
Comment (arellegue) [inline doc/snippets/Microsoft.Data.SqlClient/SqlAuthenticationProvider.xml:13]: I thought we should not be referencing System.Linq. Is it alright in this case?

**Review by arellegue — APPROVED** [review]

**Pull request merged**